airbyte-integrations/connectors/source-harvest/unit_tests/integration/test_clients.py

```diff
@@ -0,0 +1,460 @@
+                    {
+                        "clients": [
+                            {
+                                "id": 201,
+                                "name": "New Client Corp",
+                                "is_active": True,
+                                "currency": "USD",
+                                "created_at": "2024-01-02T10:00:00Z",
+                                "updated_at": "2024-01-02T10:00:00Z",
+                            }
+                        ],
+                        "per_page": 50,
+                        "total_pages": 1,
+                        "page": 1,
+                        "links": {},
+                    }
+                ),
+                status_code=200,
+            ),
+        )
+
+        # ACT: Run incremental sync
+        source = get_source(config=config, state=state)
+        catalog = CatalogBuilder().with_stream(_STREAM_NAME, SyncMode.incremental).build()
+        output = read(source, config=config, catalog=catalog, state=state)
+
+        # ASSERT: Should return only records updated since last sync
+        assert len(output.records) == 1
+        assert output.records[0].record.data["id"] == 201
+        assert output.records[0].record.data["name"] == "New Client Corp"
+
+        # ASSERT: State should be updated with the timestamp of the latest record
+        assert len(output.state_messages) > 0
+        latest_state = output.state_messages[-1].state.stream.stream_state
+        assert (
+            latest_state.__dict__["updated_at"] == "2024-01-02T10:00:00Z"
+        ), "State should be updated to the updated_at timestamp of the latest record"
+
+    @HttpMocker()
+    def test_empty_results(self, http_mocker: HttpMocker):
+        """
+        Test that connector handles empty results gracefully.
+
+        Given: An API that returns no clients
+        When: Running a full refresh sync
+        Then: The connector should return zero records without errors
+        """
+        config = ConfigBuilder().with_account_id(_ACCOUNT_ID).with_api_token(_API_TOKEN).build()
+
+        # ARRANGE: Mock empty response
+        http_mocker.get(
+            HarvestRequestBuilder.clients_endpoint(_ACCOUNT_ID, _API_TOKEN)
+            .with_per_page(50)
+            .with_updated_since("2021-01-01T00:00:00Z")
+            .build(),
+            HarvestPaginatedResponseBuilder.empty_page("clients"),
+        )
+
+        # ACT: Run the connector
+        source = get_source(config=config)
+        catalog = CatalogBuilder().with_stream(_STREAM_NAME, SyncMode.full_refresh).build()
+        output = read(source, config=config, catalog=catalog)
+
+        # ASSERT: No records but no errors
+        assert len(output.records) == 0
+
+        # ASSERT: Should have log messages indicating successful sync completion
+        log_messages = [log.log.message for log in output.logs]
+        assert any("Finished syncing" in msg for msg in log_messages)
```

Comment:
I would caution against performing assertions against any log messages because we don't make any guarantees that we will retain the same logging behavior at any point in the future.

I recommend removing this assertion as it makes our tests pretty brittle.